<launch>
  <arg name="INPUT_IMAGE"/>
  <arg name="IMAGE_TRANSPORT" default="raw"/>
  <arg name="model_file" default="$(find coral_usb)/models/mobilenet_ssd_v2_coco_quant_postprocess_edgetpu.tflite"/>
  <arg name="label_file" default="$(find coral_usb)/models/coco_labels.txt"/>
  <arg name="machine" default="c2" />
  <machine name="c1" address="c1" default="true"/>
  <machine name="c2" address="c2" env-loader="$(find coral_usb)/launch/env.sh"/>

  <node name="edgetpu_object_detector"
        pkg="coral_usb" type="edgetpu_object_detector.py"
        output="screen" respawn="true" machine="$(arg machine)">
    <remap from="~input" to="$(arg INPUT_IMAGE)" />
    <rosparam subst_value="true" >
      model_file: $(arg model_file)
      label_file: $(arg label_file)
      image_transport: $(arg IMAGE_TRANSPORT)
    </rosparam>
  </node>
</launch>
